bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1295,6 +1291,21 @@ public static boolean isSelectivelyImmutableType(BType type, Set<BType> unresolv
 
                 return isInherentlyImmutableType(recordRestType) ||
                         isSelectivelyImmutableType(recordRestType, unresolvedTypes);
+            case TypeTags.OBJECT_TYPE_TAG:
+                BObjectType objectType = (BObjectType) type;
+
+                if (!Flags.isFlagOn(objectType.flags, Flags.ABSTRACT)) {
+                    return false;
+                }
+
+                for (BField field : objectType.getFields().values()) {
+                    BType fieldType = field.type;
+                    if (!isInherentlyImmutableType(fieldType) &&
```

Comment:
IMO using !(A || B) is much readable than (!A && !B)

Reply:
I personally find `!A && !B` more readable, especially when there are few conditions. :D I find it easier to read "if fieldType is not inherently immutable and is not selectively immutable".